### PR TITLE
media-gfx/krita-5.2.2: fix the libjxl patch

### PR DESCRIPTION
**Pull request by Jamim**
_Hello @DarkDefender,_

The `libjxl` patch for version `5.1.5` is truncated, so it doesn't fix the compatibility for `5.2.2`. The full version of the patch should be used.

<details>
<summary>diff -u files/krita-5.{1.5,2.2}-libjxl-0.9.patch</summary>

```diff
--- files/krita-5.1.5-libjxl-0.9.patch  2024-04-23 01:01:23.920620040 +0300
+++ files/krita-5.2.2-libjxl-0.9.patch  2024-04-23 13:52:19.368891785 +0300
@@ -68,6 +68,26 @@
                                                        JXL_COLOR_PROFILE_TARGET_DATA,
                                                        reinterpret_cast<uint8_t *>(iccProfile.data()),
                                                        static_cast<size_t>(iccProfile.size()))) {
+@@ -657,7 +666,9 @@ JPEGXLImport::convert(KisDocument *document, QIODevice *io, KisPropertiesConfigu
+                 if (!d.m_info.uses_original_profile) {
+                     if (JXL_DEC_SUCCESS
+                         != JxlDecoderGetICCProfileSize(dec.get(),
++#if JPEGXL_NUMERIC_VERSION < JPEGXL_COMPUTE_NUMERIC_VERSION(0,9,0)
+                                                        nullptr,
++#endif
+                                                        JXL_COLOR_PROFILE_TARGET_ORIGINAL,
+                                                        &iccTargetSize)) {
+                         errFile << "ICC profile size retrieval failed";
+@@ -667,7 +678,9 @@ JPEGXLImport::convert(KisDocument *document, QIODevice *io, KisPropertiesConfigu
+                     iccTargetProfile.resize(static_cast<int>(iccTargetSize));
+                     if (JXL_DEC_SUCCESS
+                         != JxlDecoderGetColorAsICCProfile(dec.get(),
++#if JPEGXL_NUMERIC_VERSION < JPEGXL_COMPUTE_NUMERIC_VERSION(0,9,0)
+                                                           nullptr,
++#endif
+                                                           JXL_COLOR_PROFILE_TARGET_ORIGINAL,
+                                                           reinterpret_cast<uint8_t *>(iccTargetProfile.data()),
+                                                           static_cast<size_t>(iccTargetProfile.size()))) {
 -- 
 GitLab

```

</details>

Patch source:
  - https://invent.kde.org/graphics/krita/-/commit/ace7edcc

_Best regards!_

### CODE REVIEW

File: media-gfx/krita/krita-5.2.2.ebuild [hunk 1]
```diff
@@ -97,9 +97,9 @@ BDEPEND="
 PATCHES=(
 	"${FILESDIR}/${PN}-4.3.1-tests-optional.patch"
 	"${FILESDIR}/${PN}-5.1.5-sip-6.8.patch" # bug 919139
-	"${FILESDIR}/${PN}-5.1.5-libjxl-0.9.patch" # bug 922524
-	"${FILESDIR}/${PN}-5.2.2-fftw.patch"
-	"${FILESDIR}/${PN}-5.2.2-openjpeg.patch"
+	"${FILESDIR}/${P}-libjxl-0.9.patch" # bug 922524
```
Comment (DarkDefender) [inline media-gfx/krita/krita-5.2.2.ebuild:100]: Why did you remove the version information here?
Usually these are used to indicated which version the patches were introduced or started to be relevant.

Comment (Jamim) [reply media-gfx/krita/krita-5.2.2.ebuild:100]: File names are still the same, I've just shortened their representations a bit for this particular ebuild.
I believe it's a matter of taste, but [the documentation](https://devmanual.gentoo.org/ebuild-writing/misc-files/patches/) recommends using `${P}`.

> Patches are best named ${P}-what-it-does.patch

> When mirroring patches, choosing a name that will not cause conflicts is important — the ${P} prefix is highly recommended here.

Also using `${P}` forces a developer who wants to perform a version bump to check if the patch is still relevant, so they should decide whether patch might be dropped or the corresponding ebuild line should be adjusted.

And it's widely used, although `${PN}` is more common.
```shell
# pwd
/var/db/repos/DarkDefender
# ag '{P}.*.patch' | wc -l
6323
# ag '{PN}-[0-9].*.patch' | wc -l
8281
```

But of course, I can amend my commit to use `${PN}` instead of `${P}`.
Do you want me to do so?

Comment (DarkDefender) [reply media-gfx/krita/krita-5.2.2.ebuild:100]: Fair enough.
I think I'll just accept this then as I think the affected patches has made it to upstream, so we would probably have to remove them from the new point release.
